bloom-housing/bloom
# Add storybook a11y testing #142

This is a WIP to show the proof of concept. Remaining steps:

- [x] Get the tests up and running on CircleCI
- [x] Fix some a11y issues, find path forward for others. 

## Issue

Addresses #142

## Description
This setups automated a11y testing, that raises errors when storybook finds a11y errors (rather than just showing them in the UI). The goal is to incorporate a11y checks early in the development process to catch issues before they're deployed to production. This won't catch all accessibility issues, but will reduce the need for manual testing. 

The rules can be disabled on a component or story level as needed if we want to individually override them. 

This is accomplished using the path suggested in [this blog post](https://blog.atrera.com/javascript/pipeline/2021/03/10/a-guide-to-add-auto-accessibility-check-with-storybook.html), which involves bringing back the storyshots dependency, but does not require bringing back the storyshots snapshot files. 

## Type of change

- [?] This change requires a documentation update

## How Can This Be Tested/Reviewed?

- Verify that you can run `yarn test:a11y` in the ui-components dir locally
- Verify that tests are running successfully in circle-ci

## Checklist:
